src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -271,27 +283,71 @@ ErrorIfModifyQueryNotSupported(Query *queryTree)
 				continue;
 			}
 
-			if (contain_mutable_functions((Node *) targetEntry->expr))
+			if (commandType == CMD_UPDATE &&
+				contain_volatile_functions((Node *) targetEntry->expr))
 			{
-				hasNonConstTargetEntryExprs = true;
+				ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+								errmsg("functions used in UPDATE queries on distributed "
```

Comment:
Hm. As long as it qualifies the "no var argument" rule, I don't immediately see why we couldn't also allow volatile functions here?  Is this just because the number of volatile function executions might change?